eodatasets/prepare/ls_usgs_l1_prepare.py

```diff
@@ -213,8 +216,9 @@ def prepare_dataset(path):
         mtl_filename
     )
 
+    # Generate a deterministic UUID for the level 1 dataset
     return {
-        'id': str(uuid.uuid5(uuid.NAMESPACE_URL, path.as_posix())),
+        'id': str(uuid.uuid5(USGS_UUID_NAMESPACE, path.name)),
```

Comment:
I'd prefer to make it explicit that we're using the landsat product id. Perhaps use the one from the mtl? `mtl→metadata_file_info→landsat_product_id` 

There's also an (admittedly tiny) chance that they've renamed the directory, or are processing it themselves.